tests/unit/api/main.py

```diff
@@ -412,6 +412,25 @@ def test_node_register(self):
                   "password": "tapeworm"})
         api._must_find(model.Node, 'node-99')
 
+    def test_node_register_with_metadata(self):
+        api.node_register('node-99', obm={
+                  "type": "http://schema.massopencloud.org/haas/v0/obm/ipmi",
+                  "host": "ipmihost",
+                  "user": "root",
+                  "password": "tapeworm"}, metadata={
+                  "EK": "pk"})
+        api._must_find(model.Node, 'node-99')
+
+    def test_node_register_with_multiple_metadata(self):
+        api.node_register('node-99', obm={
+                  "type": "http://schema.massopencloud.org/haas/v0/obm/ipmi",
+                  "host": "ipmihost",
+                  "user": "root",
+                  "password": "tapeworm"}, metadata={
+                  "EK": "pk",
+                  "SHA256": "b5962d8173c14"})
```

Comment:
Two things:

1. The set of metadata you choose to use in the tests doesn't really matter from a functional standpoint, so I'd pick something that is at least not incorrectly self-describing. Perhaps somthing that is more obviously arbitrary just test data, e.g. `{"foo": "bar", "baz": "quux"}`.
2. Python will concatinate string adjacent string literals (just like C does), so in cases like this you can do e.g.:

```python
{
    "EK": "pk",
    "SHA256": "e3b0c44298fc1c149afbf4c8996fb924"
              "27ae41e4649b934ca495991b7852b855",
}
```